File: LeetCode/Graph/BFS/Python/Leetcode429-N叉树的层序遍历.py
class Solution:
    def levelOrder(self, root: 'Node') -> List[List[int]]:
        if not root: return []
        res,q = [],deque()
        q.append(root)
        while q:
            n,cur = len(q),[]
            for i in range(n):
                t = q.popleft()            
                cur.append(t.val)
                for ne in t.children:
                    q.append(ne)
            res.append(cur)
        return res

File: LeetCode/Graph/BFS/Golang/Leetcode429-N叉树的层序遍历.go
package sbgo

func levelOrder(root *Node) [][]int {
	res, q := [][]int{}, []*Node{root}
	if root == nil {
		return res
	}
	for len(q) != 0 {
		n, cur := len(q), []int{}
		for i := 0; i < n; i++ {
			t := q[0]
			q = q[1:]
			cur = append(cur, t.Val)
			for _, ne := range t.Children {
				q = append(q, ne)
			}
		}
		res = append(res, cur)
	}
	return res
}
